src/main/java/com/powsybl/openloadflow/network/impl/LfTieLineBranch.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.network.impl;
+
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.LimitType;
+import com.powsybl.iidm.network.TieLine;
+import com.powsybl.openloadflow.network.*;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class LfTieLineBranch extends AbstractFictitiousLfBranch {
+
+    private final TieLine tieLine;
+
+    private final Branch.Side side;
+
+    protected LfTieLineBranch(LfNetwork network, LfBus bus1, LfBus bus2, PiModel piModel, TieLine tieLine, Branch.Side side) {
+        super(network, bus1, bus2, piModel);
+        this.tieLine = tieLine;
+        this.side = side;
+    }
+
+    public static LfTieLineBranch create(TieLine tieLine, Branch.Side side, LfNetwork network, LfBus bus1, LfBus bus2) {
+        Objects.requireNonNull(tieLine);
+        Objects.requireNonNull(bus1);
+        Objects.requireNonNull(bus2);
+        double nominalV = tieLine.getTerminal(side).getVoltageLevel().getNominalV();
+        double zb = nominalV * nominalV / PerUnit.SB;
+        TieLine.HalfLine halfLine = tieLine.getHalf(side);
+        PiModel piModel = new SimplePiModel()
+                .setR(halfLine.getR() / zb)
+                .setX(halfLine.getX() / zb)
+                .setG1(halfLine.getG1() * zb)
+                .setG2(halfLine.getG2() * zb)
+                .setB1(halfLine.getB1() * zb)
+                .setB2(halfLine.getB2() * zb);
+        return new LfTieLineBranch(network, bus1, bus2, piModel, tieLine, side);
+    }
+
+    @Override
+    public String getId() {
+        return tieLine.getHalf(side).getId();
+    }
+
+    @Override
+    public boolean hasPhaseControlCapability() {
+        return false;
+    }
+
+    @Override
+    public List<LfLimit> getLimits1(LimitType type) {
```

Comment:
This seems incorrect here because for halfline 1 you can have some limits1 but for halfline 2 you can only have some limits2, no ?

Reply:
You have to use the `side` attribute here too.